docs/src/main/asciidoc/spring-cloud-stream-support.adoc

```diff
@@ -531,12 +559,139 @@ public Supplier<Message<String>> supply() {
+            LOGGER.info("New message received: '{}', partition key: {}, sequence number: {}, offset: {}, enqueued time: {}",
+                message.getPayload().get(i),
+                ((List<Object>) message.getHeaders().get(EventHubHeaders.PARTITION_KEY)).get(i),
+                ((List<Object>) message.getHeaders().get(EventHubHeaders.SEQUENCE_NUMBER)).get(i),
+                ((List<Object>) message.getHeaders().get(EventHubHeaders.OFFSET)).get(i),
+                ((List<Object>) message.getHeaders().get(EventHubHeaders.ENQUEUED_TIME)).get(i));
+        }
+
+        Checkpointer checkpointer = (Checkpointer) message.getHeaders().get(CHECKPOINTER);
+        checkpointer.success()
+                    .doOnSuccess(success -> LOGGER.info("Message '{}' successfully checkpointed", message.getPayload()))
+                    .doOnError(error -> LOGGER.error("Exception found", error))
+                    .subscribe();
+    };
+}
+
+@Bean
+public Supplier<Message<String>> supply() {
+    return () -> {
+        LOGGER.info("Sending message, sequence " + i);
+        return MessageBuilder.withPayload("\"test"+ i++ +"\"").build();
+    };
+}
+----
+
+===== Error channels
+- Consumer error channel
+
+This channel is open by default, you can handle the error message in this way:
+[source,java]
+----
+// Replace destination with spring.cloud.stream.bindings.input.destination
+// Replace group with spring.cloud.stream.bindings.input.group
+@ServiceActivator(inputChannel = "{destination}.{group}.errors")
+public void consumerError(Message<?> message) {
+    LOGGER.error("Handling customer ERROR: " + message);
+}
+----
+- Producer error channel
+
+This channel is not open by default. You need to add a configuration in your application.properties to enable it, like this:
+```
+spring.cloud.stream.default.producer.errorChannelEnabled=true
+```
+
+You can handle the error message in this way:
+[source,java]
+----
+// Replace destination with spring.cloud.stream.bindings.output.destination
+@ServiceActivator(inputChannel = "{destination}.errors")
+public void producerError(Message<?> message) {
+    LOGGER.error("Handling Producer ERROR: " + message);
+}
+----
+
 ==== Samples
 
 Please refer to link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0[azure-spring-boot-samples] for more details.
 
 === Spring Cloud Stream Binder for Azure Service Bus
 
+==== Key concepts
+The Spring Cloud Stream Binder for Azure Service Bus provides the binding implementation for the Spring Cloud Stream.
+
+This implementation uses Spring Integration Service Bus Channel Adapters at its foundation.
+
+===== Scheduled Message
+This binder supports submitting messages to a topic for delayed processing. Users can send scheduled messages with header `x-delay`
+expressing in milliseconds a delay time for the message. The message will be delivered to the respective topics after `x-delay` milliseconds.
+
+### Consumer Group
```

Comment:
Use `===` instead of `###`.

Could you please use the `Problems` feature of Intellij IDEA?
![image](https://user-images.githubusercontent.com/13167207/145496568-0f89b4ee-79bf-4d6b-848e-2c9284baf406.png)

Reply:
And seems it should be level 4 header instead of level 3 header. i.e. Use `====` instead of `===`.